elastic/eui
# [EuiProvider] Detect OS/system light vs darkmode as a default

## Summary

Nobody asked for this, but I thought I'd do it anyway! 🤠 

EUI's theme/`colorMode` currently defaults to light mode. I've added a new provider that detects the current system light/dark mode setting and if no consumer `colorMode` prop has been set/overridden, will automatically default to the OS/system setting.

There's also a few src-docs cleanup in here (tagged as such in [individual commits](https://github.com/elastic/eui/pull/8026/commits)), and I'll shortly be doing another pass of our various docs that remove/deprecate all references to Sass setup.

## QA

- Go to https://eui.elastic.co/pr_8026/#/
- Clear any saved local data for the site:
  <img width="469" alt="" src="https://github.com/user-attachments/assets/7af6fcd2-266f-4db5-af37-72d220e5cb51">
- Go to System > Appearance > Set to dark mode
  <img width="472" alt="" src="https://github.com/user-attachments/assets/8e6e4812-db18-444f-83ef-e96de9e8870b">
- [x] Refresh the page and confirm the docs site is in dark mode
- **Without refreshing the page**, go to your system appearances again and set it to light mode 
   <img width="470" alt="" src="https://github.com/user-attachments/assets/a12c4baa-8b6b-4400-8b17-8c96ecf0cf12">
- [x] Tab back to the docs site and confirm that the site has switched over automatically
- Go to the switcher and manually change the site back to light mode
- [x] Refresh the page and confirm that preferred setting is saved and EUI is always in light mode regardless of system setting

### General checklist

- Browser QA
- Docs site QA
- Code quality checklist
- Release checklist
